cmd/algorand-indexer/daemon_test.go

```diff
@@ -102,3 +109,174 @@ func TestReadGenesis(t *testing.T) {
 	assert.Contains(t, err.Error(), "readGenesis() err: EOF")
 
 }
+
+func createTempDir(t *testing.T) string {
+	dir, err := os.MkdirTemp("", "indexer")
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	return dir
+}
+
+// Make sure we output and return an error when both an API Config and
+// enable all parameters are provided together.
+func TestConfigWithEnableAllParamsExpectError(t *testing.T) {
+	indexerDataDir := createTempDir(t)
+	defer os.RemoveAll(indexerDataDir)
+	daemonConfig := &daemonConfig{}
+	daemonConfig.flags = pflag.NewFlagSet("indexer", 0)
+	daemonConfig.indexerDataDir = indexerDataDir
+	daemonConfig.enableAllParameters = true
+	daemonConfig.suppliedAPIConfigFile = "foobar"
+	err := runDaemon(daemonConfig)
+	errorStr := "not allowed to supply an api config file and enable all parameters"
+	if err.Error() != errorStr {
+		t.Fatalf("expected error %s, but got %s", errorStr, err.Error())
+	}
```

Comment:
The error validation helpers in testify used to be pretty basic, I think there are some new ones that might help with this sort of check.

Also, theres a panic here if `runDaemon` doesn't return the expected error.
```suggestion
	require.Error(t, err)    
	require.Equals(t, err.Error(), errorStr)
```

Reply:
Fixed all the tests to use `require`